### DISCUSSION TITLE
Welcome to OnePixel Discussion Board

### DISCUSSION THREAD

**Discussion by championswimmer** [championswimmer/onepixel_backend#4] [Announcements]
### Why is it called OnePixel ?

Because the URL this will be deployed it on is 1px.li 
I want to call the product as "one pixel links" - where 'one pixel' alludes to the shortness of the links. 

### Why this project exists? 

I wrote this on Twitter 
https://twitter.com/championswimmer/status/1676573222714875907 

Got some good responses
So I started making it 

https://twitter.com/championswimmer/status/1719668649416679922 
<img width="587" alt="image" src="https://github.com/championswimmer/onepixel_backend/assets/1327050/83ade4ba-da4a-4717-a3e8-2180a7840cae">

### How do I contribute? 
Go to either of these two repos 

https://github.com/championswimmer/onepixel_backend
https://github.com/championswimmer/onepixel_frontend (not made yet, will be done soon) 

### What is the current project status? 

The current immediate goal is to reach v0.1.0 
The scope is detailed here 

https://github.com/championswimmer/onepixel_backend/milestone/1

**Comment by shawshankkumar** [comment]
Hey,
Looking forward to contribute to this @championswimmer . Small question, why not go for backend and frontend in the same repo?

**Reply by championswimmer** [reply]
no particular reason - just feels cleaner that way - given they'll be on pretty different tech stacks

the backend should be able to run with 100% features even without the frontend btw - I want to make this API first in it's approach.

**Reply by shawshankkumar** [reply]
Yeah that makes sense. What are you planning on deploying this on? 
I assume netlify for vue, but what about go? @championswimmer

**Reply by championswimmer** [reply]
Railway.app for the Go backend initially 

Will do Azure/AWS later when I make it paid and on production open for everyone. @shawshankkumar

**Comment by asharma991** [comment]
Hi @championswimmer, looking forward to contribute. I really like the idea of this shortner. I hope it'll help me with learning more about Go and Vue.js.

**Comment by vinikatyal** [comment]
I have worked on Vue would love to contribute

**Reply by championswimmer** [reply]
Sure, the frontend should be started soon, once the basic v0.1 of the API is ready.

**Comment by akshay-devadiga** [comment]
Hi @championswimmer , I've been working with Vue for some time, and i would like to contribute as well

**Comment by rudrakshsisodiya** [comment]
Hi @championswimmer, It has been more than 2years on Go, love to contribute XD

**Reply by championswimmer** [reply]
Just go ahead and start contributing :)

**Comment by sbshah97** [comment]
Would be interesting to help out with Go.

**Comment by dineshyadav19** [comment]
Hi @championswimmer, Would love to contribute on the frontend part.

**Comment by manosriram** [comment]
Have worked on Go, would like to contribute to the project.
Thanks!

**Comment by sanishchirayath1** [comment]
I would love to contribute to the  front-end side!

**Comment by sptsway** [comment]
i would like to contribute too to the backend side, and also have proficeincy with Go.

**Reply by championswimmer** [reply]
@tts89 don't get me wrong, but just saying "would like to contribute" adds little value. we are here to build not to talk. go ahead and make PRs.

**Comment by paras151** [comment]
Looking forward to this, seems interesting to work on an opensource repo from an initial phase

**Comment by Jitendra-kaswa** [comment]
Hii @championswimmer,  I would like to contribute to the backend side.

**Reply by championswimmer** [reply]
@Jitendra-kaswa don't just say "I would like to contribute". Go ahead and make PRs

**Comment by mrudang-vora** [comment]
Hi @championswimmer, do you have any design document for this where we can understand the high level overview and contribute to this project?

**Reply by championswimmer** [reply]
@mrudang-vora it is too simple to require a design doc. it has 3 models, 2 controllers. if the code is not sufficient at this level of project to start contributing then i am afraid I won't be able to help much :( 

There is a OpenAPI YAML file to lay out the expectation from the API for v0.1
